### PR TITLE
Final

### LINKED ISSUE #18: Make clients final
| Q            | A
| ------------ | ---
| Bug?         | no
| New Feature? | yes
| Version      | 2.0

Clients in this package is not final at the moment. I think they are not intended to be extended, so they should be made final.

### PR DESCRIPTION
**Pull request by sagikazarmark**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #18
| License         | MIT

#### What's in this PR?

Makes some clients final, adds todos for clients becoming final in the next major version.

#### Why?

Cleans the confusion about clients being extended or not.

#### Checklist

- [x] Updated CHANGELOG.md to describe BC breaks / deprecations | new feature | bugfix

### CODE REVIEW

File: spec/FlexibleHttpClientSpec.php [hunk 1]
```diff
@@ -92,3 +88,22 @@ function it_does_not_emulate_a_client(FlexibleHttpClient $client, RequestInterfa
...
 }
+
+class CombinedHttpClient implements HttpClient, HttpAsyncClient
+{
+    /**
+     * @param RequestInterface $request
+     */
+    public function sendRequest(RequestInterface $request)
+    {
+        // TODO: Implement sendRequest() method.
```
Comment (dbu) [inline spec/FlexibleHttpClientSpec.php]: ? and why is this class embedded in the spec file?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 - Add a flexible http client providing both contract, and only emulating what's necessary
 - HTTP Client Router: route requests to underlying clients
+- TODO for existing clients to become final in 2.0
```
Comment (dbu) [inline CHANGELOG.md]: i don't think this needs to go into the changelog. the changelog should only contain things people need to act on. so maybe: "Deprecated extending client classes. They will be made final in client-common 2.0"

**Comment by dbu** [comment]
lets add an issue in the 2.0 milestone to remember to make them final and solve the TODO.

File: spec/FlexibleHttpClientSpec.php (hunk 1 above)
Comment (sagikazarmark) [inline spec/FlexibleHttpClientSpec.php]: It's a stub. We need something which implements both interfaces, and it can't be final.

Comment (dbu) [inline spec/FlexibleHttpClientSpec.php]: then lets call it StubCombinedClient and not write TODO here but just have empty body. and a class phpdoc that says this is a stub for testing.

File: src/FlexibleHttpClient.php [hunk 3]
```diff
@@ -11,7 +11,7 @@
  *
  * @author Joel Wurtz <joel.wurtz@gmail.com>
  */
-class FlexibleHttpClient implements HttpClient, HttpAsyncClient
+final class FlexibleHttpClient implements HttpClient, HttpAsyncClient
```
Comment (dbu) [inline src/FlexibleHttpClient.php]: did we want to rename this to CombinedHttpClient now?

Comment (sagikazarmark) [inline src/FlexibleHttpClient.php]: Don't know, rename is discussed in #16, although no better name came up yet IMO.

**Comment by sagikazarmark** [comment]
Didn't need the stub after all.

**Pull request merged**